src/App.tsx

```diff
@@ -14,11 +15,102 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  selectedGoods: string[],
+};
+
+export class App extends React.Component <{}, State> {
+  state: Readonly<State> = {
+    goods: goodsFromServer,
```

Comment:
You don't need to store the initial goods array in the state. You should add  variables to a state if you're gonna change them